scipy/sparse/linalg/isolve/iterative.py

```diff
@@ -147,292 +393,318 @@ def combine(fn):
                """
                )
 @non_reentrant()
-def bicg(A, b, x0=None, tol=1e-5, maxiter=None, M=None, callback=None, atol=None):
-    A,M,x,b,postprocess = make_system(A, M, x0, b)
-
-    n = len(b)
-    if maxiter is None:
-        maxiter = n*10
-
-    matvec, rmatvec = A.matvec, A.rmatvec
-    psolve, rpsolve = M.matvec, M.rmatvec
-    ltr = _type_conv[x.dtype.char]
-    revcom = getattr(_iterative, ltr + 'bicgrevcom')
-
-    get_residual = lambda: np.linalg.norm(matvec(x) - b)
-    atol = _get_atol(tol, atol, np.linalg.norm(b), get_residual, 'bicg')
-    if atol == 'exit':
-        return postprocess(x), 0
-
-    resid = atol
-    ndx1 = 1
-    ndx2 = -1
-    # Use _aligned_zeros to work around a f2py bug in Numpy 1.9.1
-    work = _aligned_zeros(6*n,dtype=x.dtype)
-    ijob = 1
-    info = 0
-    ftflag = True
-    iter_ = maxiter
-    while True:
-        olditer = iter_
-        x, iter_, resid, info, ndx1, ndx2, sclr1, sclr2, ijob = \
-           revcom(b, x, work, iter_, resid, info, ndx1, ndx2, ijob)
-        if callback is not None and iter_ > olditer:
+def bicg(A, b, x0=None, tol=1e-5, maxiter=10000, sparams=None, M=None, callback=None, quiet=0, debug=0):
```

Comment:
This looks backwards incompatible, in particular the `maxiter` default change and the removal of `atol`. You can't just change behavior like that, it will break code for existing users.

Also, why would we want to add `quiet` and `debug` keywords? Seems undesired.

Reply:
It would be great if you could undo the formatting-only changes and remove the backwards compat breaks, then it'll be much clearer what you changed. Now the diff is basically a full rewrite of the function, which makes it hard to review.

Reply:
The same comments apply to the other function signatures you changed.